azkaban-common/src/main/java/azkaban/imagemgmt/models/ImageRampup.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.models;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+
+/**
+ * This class represents rampup information for an image type.
+ */
+public class ImageRampup extends BaseModel {
+
+  // Represents rampup plan id
+  private int planId;
+  // Image version
+  private String imageVersion;
+  // Image version rampup percentage
+  private int rampupPercentage;
+  // Stability tag of the version being ramped up
+  private StabilityTag stabilityTag;
+
+  public int getPlanId() {
+    return planId;
+  }
+
+  public void setPlanId(int planId) {
+    this.planId = planId;
+  }
+
+  public String getImageVersion() {
+    return imageVersion;
+  }
+
+  public void setImageVersion(String imageVersion) {
+    this.imageVersion = imageVersion;
+  }
+
+  public int getRampupPercentage() {
+    return rampupPercentage;
+  }
+
+  public void setRampupPercentage(int rampupPercentage) {
+    this.rampupPercentage = rampupPercentage;
+  }
+
+  public StabilityTag getStabilityTag() {
+    return stabilityTag;
+  }
+
+  public void setStabilityTag(StabilityTag stabilityTag) {
+    this.stabilityTag = stabilityTag;
+  }
+
+  /**
+   * Enum representing the stability of the version being rampup. By default when an image version
+   * is picked up for rampup the version will be marked as "experimental" in the image_rampup table.
+   * During rampup if there is no issue with the version the version can be updated as "stable" in
+   * the image_rampup table. And if there are issue the verion will be updated as "unstable" in the
+   * image_rampup table.
+   */
+  public enum StabilityTag {
+    EXPERIMENTAL("experimental"),
```

Comment:
Keeping it consistent with all the existing enums. As the value is specified for other enums and that also provides option  to change the value going forward (if required).

Reply:
Not necessary in this case. I still feel that we should use only enum name for this case.

Reply:
By keeping the internal tag name it gives more flexibility and we can use short form as well. For example if we introduce DEPRECATED, we can use tag name as dep, for EXPERIMENTAL it can be "exp" or "test" Same case with others. We have options to change. The enum name is more from external point of view and it should be descriptive with full name. Also remember Arvind was mentioning but something called LTX. But when we expose this to user we need to provide full form. So going forward we might need to add more enums. Hence, I feel we should keep this format.